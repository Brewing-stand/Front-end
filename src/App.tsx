import './App.css'
import Navbar from "./components/Navbar.tsx";
import {BrowserRouter as Router, Routes, Route} from "react-router-dom";
import {Home} from "./pages/Home.tsx";
import {AuthGit} from "./pages/auth/AuthGit.tsx";
import Projects from "./pages/Projects.tsx";
import Auth from "./pages/auth/Auth.tsx";
import Settings from "./pages/user/Settings.tsx";

function App() {

  return (
      <>
          <Router>
              <Navbar/>
              <Routes>
                  <Route path="/" element={<Home/>}/>

                  <Route path="/Auth" element={<Auth/>}/>
                  <Route path="/Auth/Git" element={<AuthGit/>}/>

                  <Route path="/User/Settings" element={<Settings/>}/>

                  <Route path="/Projects" element={<Projects/>}/>
              </Routes>
          </Router>
      </>
  )
}

export default App
