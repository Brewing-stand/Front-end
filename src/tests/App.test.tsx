import {expect, it} from "vitest";
import {render} from "@testing-library/react";
import App from "../App.tsx";

it("Snapshot Navbar", () => {
    const {asFragment} = render(<App/>);
    expect(asFragment()).toMatchSnapshot();
});
