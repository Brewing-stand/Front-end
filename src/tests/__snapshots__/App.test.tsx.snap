// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot Navbar 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
     
    <ul
      class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light css-dev-only-do-not-override-161f05s"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    >
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        data-menu-id="rc-menu-uuid-test-home"
        role="menuitem"
        style="opacity: 1; order: 0;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          Home
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        role="menuitem"
        style="opacity: 1; order: 1;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          Projects
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item"
        role="menuitem"
        style="opacity: 1; order: 2; margin-left: auto; background: rgb(22, 119, 255); color: white;"
        tabindex="-1"
      >
        <span
          aria-label="login"
          class="anticon anticon-login ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="login"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M521.7 82c-152.5-.4-286.7 78.5-363.4 197.7-3.4 5.3.4 12.3 6.7 12.3h70.3c4.8 0 9.3-2.1 12.3-5.8 7-8.5 14.5-16.7 22.4-24.5 32.6-32.5 70.5-58.1 112.7-75.9 43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 32.6 32.5 58.1 70.4 76 112.5C865.7 417.8 875 464.1 875 512c0 47.9-9.4 94.2-27.8 137.8-17.8 42.1-43.4 80-76 112.5s-70.5 58.1-112.7 75.9A352.8 352.8 0 01520.6 866c-47.9 0-94.3-9.4-137.9-27.8A353.84 353.84 0 01270 762.3c-7.9-7.9-15.3-16.1-22.4-24.5-3-3.7-7.6-5.8-12.3-5.8H165c-6.3 0-10.2 7-6.7 12.3C234.9 863.2 368.5 942 520.6 942c236.2 0 428-190.1 430.4-425.6C953.4 277.1 761.3 82.6 521.7 82zM395.02 624v-76h-314c-4.4 0-8-3.6-8-8v-56c0-4.4 3.6-8 8-8h314v-76c0-6.7 7.8-10.5 13-6.3l141.9 112a8 8 0 010 12.6l-141.9 112c-5.2 4.1-13 .4-13-6.3z"
            />
          </svg>
        </span>
        <span
          class="ant-menu-title-content"
        >
          Sign in
        </span>
      </li>
      <li
        aria-hidden="true"
        class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
      >
        <div
          aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-rc-menu-more"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
    <div
      class="flex items-center justify-center min-h-screen bg-gray-50 py-10"
    >
      <div
        class="ant-card w-full max-w-2xl shadow-xl rounded-lg css-dev-only-do-not-override-161f05s"
        style="padding: 30px; background-color: rgb(255, 255, 255);"
      >
        <div
          class="ant-card-body"
        >
          <h2
            class="ant-typography text-center text-gray-800 css-dev-only-do-not-override-161f05s"
          >
            Welcome to Brewing Stand!
          </h2>
          <div
            class="ant-typography text-center text-gray-600 mb-6 css-dev-only-do-not-override-161f05s"
          >
            Hello, Guest! Please log in to get started with Brewing Stand.
          </div>
          <div
            class="flex justify-center"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-161f05s ant-btn-round ant-btn-primary ant-btn-color-primary ant-btn-variant-solid ant-btn-lg bg-gray-900 hover:bg-gray-800 text-white border-none"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <span
                  aria-label="fire"
                  class="anticon anticon-fire"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="fire"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M834.1 469.2A347.49 347.49 0 00751.2 354l-29.1-26.7a8.09 8.09 0 00-13 3.3l-13 37.3c-8.1 23.4-23 47.3-44.1 70.8-1.4 1.5-3 1.9-4.1 2-1.1.1-2.8-.1-4.3-1.5-1.4-1.2-2.1-3-2-4.8 3.7-60.2-14.3-128.1-53.7-202C555.3 171 510 123.1 453.4 89.7l-41.3-24.3c-5.4-3.2-12.3 1-12 7.3l2.2 48c1.5 32.8-2.3 61.8-11.3 85.9-11 29.5-26.8 56.9-47 81.5a295.64 295.64 0 01-47.5 46.1 352.6 352.6 0 00-100.3 121.5A347.75 347.75 0 00160 610c0 47.2 9.3 92.9 27.7 136a349.4 349.4 0 0075.5 110.9c32.4 32 70 57.2 111.9 74.7C418.5 949.8 464.5 959 512 959s93.5-9.2 136.9-27.3A348.6 348.6 0 00760.8 857c32.4-32 57.8-69.4 75.5-110.9a344.2 344.2 0 0027.7-136c0-48.8-10-96.2-29.9-140.9zM713 808.5c-53.7 53.2-125 82.4-201 82.4s-147.3-29.2-201-82.4c-53.5-53.1-83-123.5-83-198.4 0-43.5 9.8-85.2 29.1-124 18.8-37.9 46.8-71.8 80.8-97.9a349.6 349.6 0 0058.6-56.8c25-30.5 44.6-64.5 58.2-101a240 240 0 0012.1-46.5c24.1 22.2 44.3 49 61.2 80.4 33.4 62.6 48.8 118.3 45.8 165.7a74.01 74.01 0 0024.4 59.8 73.36 73.36 0 0053.4 18.8c19.7-1 37.8-9.7 51-24.4 13.3-14.9 24.8-30.1 34.4-45.6 14 17.9 25.7 37.4 35 58.4 15.9 35.8 24 73.9 24 113.1 0 74.9-29.5 145.4-83 198.4z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                Start Brewing
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
